Providers/SyncStatusResolver.swift

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Sync
+import XCGLogger
+import Deferred
+import Shared
+import Storage
+
+public enum SyncDisplayState {
+    case InProgress
+    case Good
+    case Bad(message: String?)
+    case Stale(message: String)
```

Comment:
I've renamed this to Warning since that what we display to the user. I don't think the meaning of Stale makes sense here.